v3/src/components/case-table/use-index-column.tsx

```diff
@@ -59,10 +59,48 @@ export const IndexCell = ({ caseId, index, onClick }: ICellProps) => {
     // no dependencies means we'll check/fix it after every render
   })
 
+  // Find the parent CODAP component to display the index menu above the grid
+  useEffect(() => {
+    if (cellElt && !codapComponentElt) {
+      let parent: HTMLElement | null
+      for (parent = cellElt; parent; parent = parent.parentElement) {
+        if (parent.classList.contains("codap-component")) {
+          setCodapComponentElt(parent)
+          break
+        }
+      }
+    }
+  }, [cellElt, codapComponentElt])
+
+  const handleMoveDataRow = () => {
```

Comment:
Better name would be:
```suggestion
  const handleMoveEntryRow = () => {
```